### DISCUSSION TITLE
Question about x-bind target

### DISCUSSION THREAD

**Discussion by personalsoft-fabianobonin** [alpinejs/alpine#1751] [1. Help] [answered]
Hi,

I have a custom element with 2 custom properties, "value" and "xxxxx".
Note they are not attributes, they are real properties, with getters and setters.

```javascript
<body x-data="{ prop: 'value' }">
  <my-element x-bind:value="prop" x-bind:xxxxx="prop"></my-element>

  <script>
    class MyElement extends HTMLElement {
      _value = null;
      _xxxxx = null;
      get xxxxx() { return this._xxxxx; }
      set xxxxx(xxxxx) { this._xxxxx = xxxxx; }
      get value() { return this._value; }
      set value(value) { this._value = value; }
    }
    customElements.define('my-element', MyElement);
  </script>
</body>
```

When I bind something to "value", it binds to the "value" property (see below)
When I bind something to "xxxxx", it binds to the "xxxxx" attribute (see below)

```javascript
document.querySelector('my-element').value
-> "value"

document.querySelector('my-element').getAttribute("value")
-> null

document.querySelector('my-element').xxxxx
-> null

document.querySelector('my-element').getAttribute("xxxxx")
-> "value"
```

I presume there is some special treatment when binding to "value", so my questions are:

* Is this behavior expected?
* As we have attribute binding (x-bind:[attribute]="[expression]"), shouldn't a regular bind always bind to a property? Or at least bind to a property if there is a property with that name?
* If this behavior is expected or need to be kept for compatibility reasons, is there some way I can bind to my custom property?

Fabiano

**Comment by SimoTod** [comment]
Value, style and class are handled differently, see https://github.com/alpinejs/alpine/blob/main/packages/alpinejs/src/utils/bind.js

**Reply by personalsoft-fabianobonin** [reply]
Is there some way to bind to a property instead of an attribute?

**Comment by personalsoft-fabianobonin** [comment, marked as answer]
For those who needs to bind to properties, there is a workaround in this discussion: https://github.com/alpinejs/alpine/discussions/1815
